src/renderer/pages/Staking/Overview/components/NominatorsList/ShardedList/ShardedList.tsx

```diff
@@ -0,0 +1,198 @@
+import { Fragment, ReactNode, useEffect, useMemo, useState } from 'react';
+import { ApiPromise } from '@polkadot/api';
+import { Transition } from '@headlessui/react';
+
+import { useI18n } from '@app/providers';
+import type { Asset, Explorer, Address, NominatorInfo } from '@shared/core';
+import { cnTw } from '@shared/lib/utils';
+import { FootnoteText, Checkbox, Accordion, Shimmering, Plate, Tooltip } from '@shared/ui';
+import { ShardAccount } from '@shared/core/types/account';
+import { DotStyles } from '@shared/ui/StatusLabel/common/constants';
+import { AssetBalance } from '@entities/asset';
+import { AssetFiatBalance } from '@entities/price/ui/AssetFiatBalance';
+import { useStakingData } from '@entities/staking';
+import { NominatorsItem } from '../NominatorItem/NominatorItem';
+
+type NominatorInfoWithShard = NominatorInfo<ShardAccount>;
+
+type Props = {
+  shardsStake: NominatorInfoWithShard[];
+  api?: ApiPromise;
+  era?: number;
+  asset?: Asset;
+  explorers?: Explorer[];
+  isStakingLoading: boolean;
+  onCheckValidators: (stash?: Address) => void;
+  onToggleNominator: (nominator: Address, value?: boolean) => void;
+  getContent: (stake: NominatorInfoWithShard) => ReactNode;
+};
+
+export const ShardedList = ({
+  shardsStake,
+  api,
+  era,
+  asset,
+  explorers,
+  isStakingLoading,
+  onCheckValidators,
+  onToggleNominator,
+  getContent,
+}: Props) => {
+  const { getNextUnstakingEra, hasRedeem } = useStakingData();
+
+  const { t } = useI18n();
+  const [shards, setShards] = useState<NominatorInfoWithShard[]>(shardsStake);
+  const [selectedShards, setSelectedShards] = useState<number>(0);
+  const [withdrawnShards, setWithdrawnShards] = useState<number>(0);
+  const [unstakingShards, setUnstakingShards] = useState<number>(0);
+
+  useEffect(() => {
+    setShards(shardsStake);
+  }, [shardsStake]);
+
+  const selectAllShards = (value: boolean) => {
+    const allShards = shards.map((shard) => {
+      onToggleNominator(shard.address, value);
+      setSelectedShards(value ? shards.length : 0);
+
+      return { ...shard, isSelected: value };
+    });
+
+    setShards(allShards);
+  };
+
+  const selectShard = (nominator: string, isChecked: boolean) => {
+    setSelectedShards((prev) => (isChecked ? prev + 1 : prev - 1));
+    onToggleNominator(nominator);
+  };
+
+  const getShards = (shardsStake: NominatorInfoWithShard[]) => {
+    if (!shards.length) return;
+
+    return shards.map((shard: NominatorInfoWithShard) => {
+      if (hasRedeem(shard.unlocking, era)) {
+        setWithdrawnShards((prev) => prev + 1);
+      }
+      if (getNextUnstakingEra(shard.unlocking, era)) {
+        setUnstakingShards((prev) => prev + 1);
+      }
+
+      return (
+        <li key={shard.account.id}>
```

Comment:
there's a bit of prop drilling but to fix it we need to refactor `Overveiw` component first